### PR TITLE
[DOC release] document {{mut}} helper

### PR DESCRIPTION
**Pull request by EricSchank**
Add docs for `{{mut}}` helper

### CODE REVIEW

**Comment by rwjblue** [comment]
:heart: THANK YOU!

File: packages/ember-htmlbars/lib/keywords/mut.js [hunk 1]
```diff
@@ -9,6 +14,39 @@ import { INVOKE, ACTION } from 'ember-routing-htmlbars/keywords/closure-action';
 
 export let MUTABLE_REFERENCE = symbol('MUTABLE_REFERENCE');
 
+/**
+  The `mut` helper lets you __clearly specify__ that a child `Component` can update the
+  (mutable) value passed to it, which will __change the value of the parent compnent__.
```
Comment (steffenbrem) [inline packages/ember-htmlbars/lib/keywords/mut.js]: The last word has a typo `compnent__` must be `component__`.

Comment (rwjblue) [inline packages/ember-htmlbars/lib/keywords/mut.js]: @steffenbrem - Mind submitting a fixing PR?

Comment (steffenbrem) [inline packages/ember-htmlbars/lib/keywords/mut.js]: @rwjblue I've opened a PR for this: #12031

**Comment by mixonic** [comment]
It was my understanding this was explicitly *not* documented or public. /cc @wycats

**Comment by rwjblue** [comment]
@mixonic - We can make it private if needed, docs are still better than no docs.

**Pull request merged**